opal/datatype/opal_datatype_get_count.c

```diff
@@ -143,3 +141,54 @@ int32_t opal_datatype_set_element_count( const opal_datatype_t* datatype, size_t
     }
 }
 
+/**
+ * Compute the array of counts of the predefined datatypes contained in
+ * the datatype. We have no simple way to create this array, as we only
+ * sporadically need it (when we deal with heterogeneous environments or
+ * when we use get_element_count). Thus, we will pay the cost once per
+ * datatype, but we will only update this array if/when needed.
+ */
+int opal_datatype_compute_ptypes( opal_datatype_t* datatype )
```

Comment:
Thanks for noticing. The same pattern appears in several location, I fixed them all. Please review the updated version.